File: Readio/src/main/java/ninegle/Readio/publisher/mapper/PublisherMapper.java
package ninegle.Readio.publisher.mapper;

import java.util.ArrayList;
import java.util.List;

import ninegle.Readio.publisher.domain.Publisher;
import ninegle.Readio.publisher.dto.PublisherListResponseDto;
import ninegle.Readio.publisher.dto.PublisherResponseDto;

public class PublisherMapper {

	public static PublisherResponseDto toResponseDto(Publisher publisher) {
		return PublisherResponseDto.builder()
			.id(publisher.getId())
			.name(publisher.getName())
			.build();
	}

	public static PublisherListResponseDto toListResponseDto(List<Publisher> publishers) {
		List<PublisherResponseDto> publisherDtos = new ArrayList<>();

		for (Publisher publisher : publishers) {
			publisherDtos.add(toResponseDto(publisher));
		}
		return PublisherListResponseDto.builder()
			.publishers(publisherDtos)
			.build();

	}

}

File: Readio/src/main/java/ninegle/Readio/publisher/mapper/PublisherMapper.kt
package ninegle.Readio.publisher.mapper

import ninegle.Readio.publisher.domain.Publisher
import ninegle.Readio.publisher.dto.PublisherListResponseDto
import ninegle.Readio.publisher.dto.PublisherResponseDto



fun Publisher.toResponseDto(): PublisherResponseDto {
    return PublisherResponseDto(
        id = this.id!!,
        name = this.name
    )
}

fun MutableList<Publisher>.toListResponseDto(): PublisherListResponseDto {
    val publisherDtos: MutableList<PublisherResponseDto> = this.map { it.toResponseDto() }.toMutableList()

    return PublisherListResponseDto(
        publishers = publisherDtos
    )
}

